include/wx/generic/imaglist.h

```diff
@@ -10,48 +10,57 @@
 #ifndef _WX_IMAGLISTG_H_
 #define _WX_IMAGLISTG_H_
 
+#include "wx/bitmap.h"
 #include "wx/gdicmn.h"
+#include "wx/vector.h"
 
 class WXDLLIMPEXP_FWD_CORE wxDC;
-class WXDLLIMPEXP_FWD_CORE wxBitmap;
 class WXDLLIMPEXP_FWD_CORE wxIcon;
 class WXDLLIMPEXP_FWD_CORE wxColour;
 
 
 class WXDLLIMPEXP_CORE wxGenericImageList: public wxObject
 {
 public:
-    wxGenericImageList() { m_width = m_height = 0; }
+    wxGenericImageList() { }
     wxGenericImageList( int width, int height, bool mask = true, int initialCount = 1 );
     virtual ~wxGenericImageList();
     bool Create( int width, int height, bool mask = true, int initialCount = 1 );
-    bool Create();
 
     virtual int GetImageCount() const;
     virtual bool GetSize( int index, int &width, int &height ) const;
-    virtual wxSize GetSize() const { return wxSize(m_width, m_height); }
+    virtual wxSize GetSize() const { return m_size; }
 
     int Add( const wxBitmap& bitmap );
     int Add( const wxBitmap& bitmap, const wxBitmap& mask );
     int Add( const wxBitmap& bitmap, const wxColour& maskColour );
     wxBitmap GetBitmap(int index) const;
     wxIcon GetIcon(int index) const;
-    bool Replace( int index, const wxBitmap &bitmap );
-    bool Replace( int index, const wxBitmap &bitmap, const wxBitmap& mask );
+    bool Replace( int index,
+                  const wxBitmap& bitmap,
+                  const wxBitmap& mask = wxNullBitmap );
     bool Remove( int index );
     bool RemoveAll();
 
     virtual bool Draw(int index, wxDC& dc, int x, int y,
               int flags = wxIMAGELIST_DRAW_NORMAL,
               bool solidBackground = false);
 
-    // Internal use only
-    const wxBitmap *GetBitmapPtr(int index) const;
+#if WXWIN_COMPATIBILITY_3_0
+    wxDEPRECATED_MSG("Don't use this unportable and doing nothing overload")
+    bool Create() { return true; }
```

Comment:
Seems like this should be OSX-only, as it doesn't currently exist anywhere else. Also, the message is rather awkward English, I would suggest something like "Don't use this overload, it's not portable and does nothing".

Reply:
It does exist in the [generic version](https://github.com/wxWidgets/wxWidgets/blob/v3.0.0/include/wx/generic/imaglist.h#L28).

I'll fix the deprecation message, thanks.